dissect/target/plugins/os/windows/wua_history.py

```diff
@@ -0,0 +1,879 @@
+    stores historical data of performed updates.
+
+    References:
+        - https://github.com/libyal/esedb-kb/blob/main/documentation/Windows%20Update.asciidoc
+        - https://www.nirsoft.net/articles/extract-windows-updates-list-external-drive.html
+    """
+
+    DATASTORE_PATH = "sysvol/windows/softwaredistribution/datastore/datastore.edb"
+    DATASTORE_UPDATE_TABLE = "tbHistory"
+
+    def __init__(self, target):
+        super().__init__(target)
+
+        self._datastore = None
+        datastore_path = target.fs.path(self.DATASTORE_PATH)
+        if datastore_path.exists():
+            try:
+                self._datastore = EseDB(datastore_path.open())
+            except Error as e:
+                self.target.log.warning("Error opening Windows DataStore database", exc_info=e)
+
+    def check_compatible(self) -> None:
+        if not self._datastore:
+            raise UnsupportedPluginError("No Windows DataStore found")
+
+    def get_table_records(self) -> Iterator[dict[str, RecordValue]]:
+        try:
+            table = self._datastore.table(self.DATASTORE_UPDATE_TABLE)
+        except KeyError:
+            return None
+
+        for record in table.records():
+            record_data = {}
+            for column in table.columns:
+                value = record.get(column.name)
+
+                if column.name == "Date":
+                    value = oatimestamp(value)
+                elif column.name in ("ServerId", "UpdateId"):
+                    value = decode_guid(value[:16]).encode("ascii")
+                elif column.name in ("MappedResult", "UnmappedResult"):
+                    value = hex(int(value) & 0xFFFFFFFF)
+                    if column.name == "MappedResult":
+                        if value in WUA_SUCCESS_CODES_MAP:
+                            record_data["result"] = "Succeeded"
+                            record_data["result_string"] = WUA_SUCCESS_CODES_MAP[value][0]
+                            record_data["result_description"] = WUA_SUCCESS_CODES_MAP[value][1]
+                        elif value in WUA_ERROR_CODES_MAP:
+                            record_data["result"] = "Failed"
+                            record_data["result_string"] = WUA_ERROR_CODES_MAP[value][0]
+                            record_data["result_description"] = WUA_ERROR_CODES_MAP[value][1]
+                        else:
+                            record_data["result"] = "Unknown"
+                            record_data["result_string"] = "Unknown error"
+                            record_data["result_description"] = "Unknown error"
+                elif column.name == "Title":
+                    kb = re.search(r"(KB.[0-9]*)", value)
+                    if kb:
+                        record_data["kb"] = kb.group(0)
+                record_data[column.name] = value
+            yield record_data
+
+    @export(record=WuaHistoryRecord)
+    def wua_history(self):

```

Comment:
Can document the contents of the records that are yielded here? Similar to other plugins, so it shows up in `target-query -f wua_history -h`.  

Also, can you be a little bit more verbose in what you can accomplish as an analist with this plugin here? A good start is using your PR comment as inspiration ;)

Reply:
I'd happily do it, but can you give a example plugin of what you mean? I can't quickly find one where that was done

Reply:
You could use this as in example: https://github.com/fox-it/dissect.target/blob/11afdf0e06bec1a25e8bf820bc26564e107a2853/dissect/target/plugins/os/windows/log/schedlgu.py#L124-L142